modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts

```diff
@@ -568,7 +568,7 @@ function DOMUtils(doc: Document, settings: Partial<DOMUtilsSettings> = {}): DOMU
     }
 
     while (node) {
-      if (node === root || !node.nodeType || node.nodeType === 9) {
+      if (node === root || !node.nodeType || NodeType.isDocument(node) || NodeType.isDocumentFragment(node)) {
```

Comment:
You don't need the `!node.nodeType` here anymore.

Reply:
`NodeType.*` will return `false` if there's no `nodeType` (and we need a `true` here), so I believe we unfortunately still need this check.

I'm open to re-working this logic so it's clearer

Reply:
`node.nodeType` can never be falsy though, as it's always a positive integer. As such it could never run into the case you mentioned, which is why I said it could be removed.

Reply:
Discussed on Slack, there is an edge case where `nodeType` can be undefined due to a bug elsewhere. Adjusting this so it's more explicit and I'll reference the new JIRA